desc/particles.py

```diff
@@ -1242,3 +1281,195 @@ def _intfun_wrapper(
         event=event,
         throw=throw,
     )
+
+
+class FourierChebyshevField(IOAble):
+    """Convenience class for fitting and evaluating fields.
+
+    This class is intended to be used during particle tracing to reduce overhead
+    of creating transforms.
+
+    Parameters
+    ----------
+    L : int
+        Maximum order of the Chebyshev polynomial to be used in the radial direction.
+    M : int
+        Maximum order of the Fourier series to be used in the poloidal direction.
+    N : int
+        Maximum order of the Fourier series to be used in the toroidal direction.
+    """
+
+    _static_attrs = ["L", "M", "N", "data_keys"]
+
+    def __init__(self, L, M, N):
+        self.L = L
+        self.M = M
+        self.N = N
+
+    def build(self, eq):
+        """Build the constants for fit.
+
+        During optimization, equilibrium field changes, however, the same transforms
+        can be used to get the fit faster. This method creates the grid and transforms
+        to be used during the fitting procedure.
+
+        Parameters
+        ----------
+        eq : Equilibrium
+            Equilibrium to be used to get transforms.
+
+        """
+        self.data_keys = ["B", "grad(|B|)", "e^rho", "e^theta", "e^zeta"]
```

Comment:
`e^theta` diverges on axis so it can't be fit well, you can probably use `e^theta*sqrt(g)`

Reply:
also just to note, `e^rho` and `e^theta*sqrt(g)` don't scale like rho^m near axis so shouldn't be fit to a zernike basis